pkg/resource/user_group/custom_set_output.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package user_group
+
+import (
+	"context"
+	svcapitypes "github.com/aws-controllers-k8s/elasticache-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (rm *resourceManager) CustomDescribeUserGroupsSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *elasticache.DescribeUserGroupsOutput,
+	ko *svcapitypes.UserGroup,
+) (*svcapitypes.UserGroup, error) {
+	if len(resp.UserGroups) == 0 {
+		return nil, ackerr.NotFound
+	}
+	elem := resp.UserGroups[0]
+	rm.customSetOutput(elem.UserIds,
+		elem.Engine,
+		elem.Status,
+		ko)
+	return ko, nil
+}
+
+func (rm *resourceManager) CustomCreateUserGroupSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *elasticache.CreateUserGroupOutput,
+	ko *svcapitypes.UserGroup,
+) (*svcapitypes.UserGroup, error) {
+	rm.customSetOutput(resp.UserIds,
+		resp.Engine,
+		resp.Status,
+		ko)
+	return ko, nil
+}
+
+func (rm *resourceManager) customSetOutput(
+	userIds []*string,
+	engine *string,
+	status *string,
+	ko *svcapitypes.UserGroup,
+) {
+	if userIds != nil {
+		ko.Spec.UserIDs = userIds
+	}
+
+	if engine != nil {
+		ko.Spec.Engine = engine
+	}
+
+	if ko.Status.Conditions == nil {
+		ko.Status.Conditions = []*ackv1alpha1.Condition{}
+	}
```

Comment:
The above is not necessary. Line 144 and line 237 in sdk.go will have already done this for you.